ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -3258,8 +3260,19 @@ public Partition createPartition(Table tbl, Map<String, String> partSpec) throws
 
   public org.apache.hadoop.hive.metastore.api.Partition getPartition(String dbName, String tableName,
```

Comment:
The only place where this method is called currently in the code is in AlterTableAddPartitionOperation.java. Instead of introducing an addtional RPC, perhaps a better way is to send the Table (ql.metadata.Table) object in this method so that we can pass the tableId directly without getting the table from metastore.

Reply:
Sure, makes sense.